proto/regen/ecocredit/v1beta1/state.proto

```diff
@@ -0,0 +1,225 @@
+        primary_key: {fields: "class_id,issuer"}
+    };
+
+    // class_id is the unique ID of credit class.
+    string class_id = 1;
+
+    // issuer is the approved issuer of the credit class.
+    string issuer = 2;
+}
+
+// ProjectInfo represents the high-level on-chain information for a project.
+message ProjectInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 3,
+        primary_key: {fields: "project_id"}
+        index: {id: 1 fields: "issuer"}
+        index: {id: 2 fields: "class_id"}
+    };
+
+    // project_id is the unique ID of the project.
+    string project_id = 1;
+
+    // class_id is the unique ID of credit class for this project.
+    string class_id = 2;
+
+    // issuer is the issuer of the credit batches for this project.
+    string issuer = 3;
+
+    // project_location is the location of the project.
+    // Full documentation can be found in MsgCreateProject.project_location.
+    string project_location = 4;
+
+    // metadata is any arbitrary metadata attached to the project.
+    bytes metadata = 5;
+}
+
+// BatchInfo represents the high-level on-chain information for a credit batch.
+message BatchInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 4,
+        primary_key: {fields: "id", auto_increment: true}
+        index: {fields: "batch_denom", unique: true}
+        index: {id: 1 fields: "project_id"}
+        index: {id: 2 fields: "start_date"}
+    };
+
+    // id is a an incrementing integer to identify the batch
+    uint64 id = 1;
+
+    // project_id is the unique ID of the project this batch belongs to.
+    string project_id = 2;
+
+    // batch_denom is the unique ID of credit batch.
+    string batch_denom = 3;
+
+    // metadata is any arbitrary metadata attached to the credit batch.
+    bytes metadata = 5;
+
+    // start_date is the beginning of the period during which this credit batch
+    // was quantified and verified.
+    google.protobuf.Timestamp start_date = 6 [ (gogoproto.stdtime) = true ];
+
+    // end_date is the end of the period during which this credit batch was
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 7 [ (gogoproto.stdtime) = true ];
+}
+
+// CreditType defines the measurement unit/precision of a certain credit type
+// (e.g. carbon, biodiversity...)
+message CreditType {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 5,
+        primary_key: {fields: "abbreviation"}
+    };
+
+    // the type of credit (e.g. carbon, biodiversity, etc)
+    string name = 1;
+
+    // abbreviation is a 1-3 character uppercase abbreviation of the CreditType
+    // name, used in batch denominations within the CreditType. It must be unique.
+    string abbreviation = 2;
+
+    // the measurement unit (e.g. kg, ton, etc)
+    string unit = 3;
+
+    // the decimal precision
+    uint32 precision = 4;
+}
+
+// CreditTypeSequence stores and increments the sequence number for credit types
+message CreditTypeSequence {
```

Comment:
What are these sequences being used for?

Reply:
i believe it was added in the case we want multiple types of "carbon" credits with different parameters, i.e carbon but measured in a different unit. in which case the abbreviation would still contain "C" and then a sequence for each additional type C0, C01, C02 etc

Reply:
that doesn't make sense 🤔 . I think we should nix it. the abbreviation is the primary key and this is controlled by governance

Reply:
would it still be auto-generated then?

Reply:
credit type was never meant to be auto-generated. it is chosen by governance. i'm confused by where this seq came from tbh

Reply:
no, the abbreviation is chosen by governance. just `C` for instance. let's fix it

Reply:
https://github.com/regen-network/regen-ledger/pull/700/commits/a40947c8ec4f81850b3491bb44aab356be08f474

Reply:
added an issue to remove the sequencing logic from the ecocredit module https://github.com/regen-network/regen-ledger/issues/707